packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -123,32 +125,52 @@ async function instantiateModule(
     }
   }
 
+  const ssrImportMeta = { url }
+  const ssrArguments: Record<string, any> = {
+    global: context.global,
+    [ssrModuleExportsKey]: ssrModule,
+    [ssrImportMetaKey]: ssrImportMeta,
+    [ssrImportKey]: ssrImport,
+    [ssrDynamicImportKey]: ssrDynamicImport,
+    [ssrExportAllKey]: ssrExportAll
+  }
+
+  let ssrModuleImpl = isProduction
+    ? result.code + `\n//# sourceURL=${mod.url}`
+    : `(0,function(${Object.keys(ssrArguments)}){\n${result.code}\n})`
+
+  const { map } = result
+  if (map?.mappings) {
+    if (mod.file) {
+      await injectSourcesContent(map, mod.file, moduleGraph)
+    }
+
+    ssrModuleImpl += `\n` + convertSourceMap.fromObject(map).toComment()
```

Comment:
The inline sourcemap is appended in both production and development, as  `new Function` still handles sourcemaps somewhat, but `vm.runInThisContext` actually respects breakpoints.